pkg/client/restclient/config.go

```diff
@@ -45,11 +45,11 @@ const (
 // Config holds the common attributes that can be passed to a Kubernetes client on
 // initialization.
 type Config struct {
-	// Host must be a host string, a host:port pair, or a URL to the base of the apiserver.
-	// If a URL is given then the (optional) Path of that URL represents a prefix that must
-	// be appended to all request URIs used to access the apiserver. This allows a frontend
-	// proxy to easily relocate all of the apiserver endpoints.
-	Host string
+	// Hosts must be a slice of host strings, host:port pairs, or a URLs to the base of the
+	// apiservers. If a URL is given then the (optional) Path of that URL represents a prefix
+	// that must be appended to all request URIs used to access the apiserver. This allows a
+	// frontend proxy to easily relocate all of the apiserver endpoints.
+	Hosts []string
```

Comment:
@smarterclayton i am not sure that it will affect amount of lines changed, i will still have to pass those AlternateHosts all over the place, but i will check it. What would be the other reason to change this? I think that it only introduces ambiguity, all those api-servers are equal and which one i, as an operator, should select, what it will affect

@justinsb you mean that this restclient can be used by external apps?